python-package/xgboost/spark/core.py

```diff
@@ -755,7 +758,12 @@ def _fit(self, dataset):
             k: v for k, v in train_call_kwargs_params.items() if v is not None
         }
         dmatrix_kwargs = {k: v for k, v in dmatrix_kwargs.items() if v is not None}
-        use_qdm = booster_params.get("tree_method", None) in ("hist", "gpu_hist")
+
+        if self.isDefined(self.use_quantile_dmatrix):
+            use_qdm = self.getOrDefault(self.use_quantile_dmatrix)
+        else:
+            use_qdm = CUDF_INSTALLED and \
```

Comment:
Can we assume all workers have the same python env? Otherwise, we have a rabbit hole.

Reply:
Then which logic shall we use ?

Only when all workers has CUDF installed, then enable `use_qdm` ?

Reply:
@trivialfis I think independent parameter is added for the case: different worker has different python environment .

Reply:
please don't check cudf in the driver side. Even driver side doesn't have cudf, workers still can have cudf installed and use the QDM.

Reply:
@wbo4958 Suppose some worker has cudf and some does not, If some worker use QDM, some do not, what will happen ?

Reply:
let's continue the discussion in https://github.com/dmlc/xgboost/pull/8469#pullrequestreview-1183596175

Reply:
Replied https://github.com/dmlc/xgboost/pull/8469#issuecomment-1319468447